python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -791,3 +812,51 @@ def info_string(self):
         lm_summary = self.load_metrics.summary()
         autoscaler_summary = self.summary()
         return "\n" + format_info_string(lm_summary, autoscaler_summary)
+
+
+class NodeTracker:
+    """Map nodes to their corresponding logs.
+
+    We need to be a little careful here. At an given point in time, node_id <->
+    ip can be interchangeably use, but the node_id -> ip relation is not
+    bijective _across time_ since IP addresses can be reused. Therefore, we
+    should treat node_id as the only unique identifier.
+
+    """
+
+    def __init__(self, log_dir, process_runner):
+        self.log_dir = log_dir
+        self.process_runner = process_runner
+
+        # Mapping from node_id -> (ip, node type, stdout_path, process runner)
+        # TODO(Alex): In the spirit of statelessness, we should try to load
+        # this mapping from the filesystem. _Technically_ this tracker is only
+        # used for "recent" failures though, so remembering old nodes is a best
+        # effort, therefore it's already correct
+        self.node_mapping = {}
+
+    def get_or_create_process_runner(self, node_id, ip=None, node_type=None):
+        if node_id not in self.node_mapping:
+            stdout_name = f"{node_id}.out"
```

Comment:
please make it .log

Reply:
No, .log implies structured output. This is raw stdout and follows convention. For example, see the worker output files in /tmp/ray/session_latest/logs/worker*.out